test/util/test_path_provider.py

```diff
@@ -0,0 +1,173 @@
+import os
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+
+class TestPathProviderBase(ABC):
+    """
+        used to generate the files path's in all tests
+        in particular this file is helping with the integration between running test_integration script
+        and the integration_test_runner.
+        all test path's should go through here.
+        in the future should use the constants module to be more stable.
+        we use abstract class so that every enviorment could easily setup it's own files structure.
+    """
+
+    def __init__(self, root_dir, file_helper_dir):
+        self.ROOT_DIR = root_dir
+        self.HELPERS_FILES_DIR = file_helper_dir
+
+    @property
+    def conjurrc_path(self):
+        return os.path.join(self.ROOT_DIR, self.conjurrc_file_name)
+
+    @property
+    def netrc_path(self):
+        return os.path.join(self.ROOT_DIR, self.netrc_file_name)
+
+    @property
+    def pem_file_path(self):
+        return os.path.join(self.ROOT_DIR, self.pem_file_name)
+
+    @property
+    def conjurrc_file_name(self):
+        return '.conjurrc'
+
+    @property
+    @abstractmethod
+    def test_conjurrc_file_path(self):
+        raise NotImplemented()
+
+    @property
+    def netrc_file_name(self):
```

Comment:
same goes here. just `netrc`

Reply:
just to make sure netrc_file_name->netrc?

Reply:
yes if it doesnt conflict with imports